### PR TITLE
(FACT-2247) Add networking fact for linux

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: lib/facts/linux/dhcp_servers.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Facts
+  module Linux
+    class DhcpServers
+      FACT_NAME = 'dhcp_servers'
+
+      def call_the_resolver
+        fact_value = construct_addresses_hash
+        Facter::ResolvedFact.new(FACT_NAME, !fact_value || fact_value.empty? ? nil : fact_value, :legacy)
```
Comment (IrimieBogdan) [inline lib/facts/linux/dhcp_servers.rb:10]: I would extract `!fact_value || fact_value.empty? ? nil : fact_value` in a variable and pass it as argument in order to make it easier to read.

Comment (oanatmaria) [reply lib/facts/linux/dhcp_servers.rb]: done

File: lib/facts/linux/scope6_interfaces.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      TYPE = :legacy
+
+      def call_the_resolver
+        arr = []
+        result = {}
+        interfaces = Facter::Resolvers::NetworkingLinux.resolve(:interfaces)
+        primary = Facter::Resolvers::NetworkingLinux.resolve(:primary_interface)
+        interfaces&.each { |interface_name, info| result["scope6_#{interface_name}"] = info[:scope6] if info[:scope6] }
+
+        result.each { |fact, value| arr << Facter::ResolvedFact.new(fact, value, :legacy) }
```
Comment (IrimieBogdan) [inline lib/facts/linux/scope6_interfaces.rb:16]: Maybe rename `arr` to `resolved_facts`

Comment (IrimieBogdan) [reply lib/facts/linux/scope6_interfaces.rb:16]: Same for `result` ... it should be `scope6_interfaces`

File: lib/resolvers/networking_linux_resolver.rb [hunk 3]
```diff
@@ -39,48 +43,103 @@ def retrieve_interface_info
 
             fill_ip_v4_info!(ip_tokens, interfaces)
             fill_io_v6_info!(ip_tokens, interfaces)
+            find_dhcp(ip_tokens, interfaces)
```
Comment (IrimieBogdan) [inline lib/resolvers/networking_linux_resolver.rb:46]: `find_dhcp!(...)`

Comment (oanatmaria) [reply lib/resolvers/networking_linux_resolver.rb]: done

File: lib/facts/linux/scope6_interfaces.rb (hunk 2 above)
Comment (oanatmaria) [reply lib/facts/linux/scope6_interfaces.rb]: done

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**